### DISCUSSION TITLE
Pin ClickHouse Cloud version

### DISCUSSION THREAD

**Discussion by adriangb** [ClickHouse/ClickHouse#56465] [Q&A]
Is it possible to pin the version of ClickHouse used in ClickHouse Cloud? We are using an HTTP client to communicate with ClickHouse and found that some version bump started sending back error data as JSON instead of plain text which broke our system. We found out because our CI runs started pulling a newer version of the docker image so nothing was impacted in production but it made us think about changes to the production version. I can dig up the details of which version bump caused it and what the issue was but that will take some bisecting and I don't think it's super relevant to the solution.

We understand that breaking changes are inevitable, even if they are not intentional, so we'd like to just pin our CHC version and do upgrades manually after testing. Is this possible?

**Comment by tsolodov** [comment]
Hi @adriangb , could you please create Support Case for this problem?

**Reply by tbragin** [reply]
Looks like the ClickHouse Support team has already proactively created a support case for this user to discuss this further and get to the bottom of incompatibility. Hope to hear back soon!

**Reply by adriangb** [reply]
Yes, they did. I'm not sure why that is the preferred method of communication, the discussion could have been public and benefitted all ClickHouse users.
